### DISCUSSION TITLE
How do I assign a phantom Camera to an Area2D and set its limits?

### DISCUSSION THREAD

**Discussion by itzjeppy** [ramokz/phantom-camera#299] [Question / Help Wanted] [answered]
So I am making a 2D platformer levels, and the levels are in a "Room" system like Megaman or Metroid, and each room can be of various shapes and sizes, and I use an area2d with rectangle collision box to mark each room's boundaries. So the thing is that when I move from one room to another room, there would be a tween transition between the rooms like in the above mentioned games, so I want the camera to limit to each room's Area2d, how would I do that, and how would I assign a new phantomcamera to each room?

**Comment by ramokz** [comment]
There's a `Limit` property for each `PCam2D` node, where you can assign a `Limit Node` as either a `CollisionShape2D` (as a child node of an `Area2D`) or a `TileMap`.

There is a 2D example scene called `2d_limit_example_scene.tscn` inside the `examples` directory of the addon that demonstrates it.

**Reply by itzjeppy** [reply]
It works when follow mode is none, but when I set follow mode to simple, the camera overrides the limits, no longer following the limits I set, how to fix it??

**Reply by ramokz** [reply]
Would you be able to share either a minimal reproduction project or a screen capture?
Am not able to replicate the issue you're describing.

**Reply by itzjeppy** [reply]
[vapormind alpha.zip](https://github.com/ramokz/phantom-camera/files/15319899/vapormind.alpha.zip)


Here's my project

**Reply by ramokz** [reply, marked as answer]
Found the issue.

It doesn't work because you're only assigning the name of the Node in the `camera.set_limit_target("CollisionShape2D")` inside `ChangeCamera.gd`.To get it to work, the parameter needs to be a full `NodePath`. When supplying an invalid `NodePath` value, it basically resets the limit that you've applied from the inspector. It probably should return an error and prevent resetting the limit, so will add that in as an update soon.

A working code example would then look like:
![image](https://github.com/ramokz/phantom-camera/assets/5159399/d3ee6655-1dbc-4565-b610-0dd6efc81a48)

**Reply by itzjeppy** [reply]
Thank you! I didn't know get_path() was a function in the first place lol
